deployment_metadata/deployment_metadata.go

```diff
@@ -0,0 +1,64 @@
+package deployment_metadata
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	orchestrationTech = "Operator"
+	argumentPrefix    = `--set-deployment-metadata=`
+
+	keyOperatorScriptVersion = "script_version"
+	keyContainerImageVersion = "container_image_version"
+	keyOrchestratorID        = "orchestrator_id"
+	keyOrchestrationTech     = "orchestration_tech"
+)
+
+type DeploymentMetadata struct {
+	OperatorScriptVersion string
+	OrchestratorID        string
+	ContainerImageVersion string
+	OrchestrationTech     string
+}
+
+func NewDeploymentMetadata(operatorScriptVersion string, orchestratorId string, containerImageVersion string) *DeploymentMetadata {
```

Comment:
```suggestion
func NewDeploymentMetadata(operatorScriptVersion string, orchestratorID string, containerImageVersion string) *DeploymentMetadata {
```

Also, we don't need to provide the OneAgent version to our environment, the agents themselves send them already. And if, as I presume it's the case, if used for OneAgent DaemonSets, it can also interfere OneAgent updates where Pods may end up being restarted twice.

In fact I think I just ran out into this bug.

As for the script version, you can also just use version.Version directly.